File: app/src/main/java/com/bigfriedicecream/recipes/views/RecipeListFragment.java
package com.bigfriedicecream.recipes.views;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bigfriedicecream.recipes.R;
import com.bigfriedicecream.recipes.interfaces.IRecipeListContract;
import com.bigfriedicecream.recipes.models.RecipeResponseModel;
import com.bigfriedicecream.recipes.presenters.RecipeListPresenter;

import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class RecipeListFragment extends Fragment implements IRecipeListContract.View {

    private IRecipeListContract.Presenter presenter;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        presenter = new RecipeListPresenter(this);
        return inflater.inflate(R.layout.fragment_recipe_list, container, false);
    }

    @Override
    public void onStart() {
        super.onStart();
        presenter.load(getContext());
    }

    public void render(Map<String, Object> recipeList) {
        for (Map.Entry<String, Object> entry : recipeList.entrySet()) {
            String key = entry.getKey();
            Object value = entry.getValue();
            System.out.println(key);
            System.out.println(value);
        }
    }
}

File: Code/app/src/main/java/com/twobrothers/overcooked/recipelist/RecipeListFragment.kt
package com.twobrothers.overcooked.recipelist

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.twobrothers.overcooked.R
import com.twobrothers.overcooked.interfaces.IRecipeListContract
import com.twobrothers.overcooked.models.RecipeModel
import com.twobrothers.overcooked.presenters.RecipeListPresenter
import com.google.gson.Gson
import com.twobrothers.overcooked.BuildConfig
import com.twobrothers.overcooked.app.ApiService
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.plugins.RxJavaPlugins
import io.reactivex.rxkotlin.subscribeBy
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.fragment_recipe_list.view.*
import org.reactivestreams.Subscriber
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

import java.util.HashMap
import io.reactivex.disposables.CompositeDisposable





class RecipeListFragment : Fragment() {

    val mDisposable = CompositeDisposable()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val retrofit = Retrofit.Builder()
                .baseUrl(BuildConfig.API_ENDPOINT)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build()

        val api = retrofit.create(ApiService::class.java)

        mDisposable.add(
                api.getRecipes()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeBy(
                        onNext = { println(it.recipes) },
                        onError =  { it.printStackTrace() },
                        onComplete = { println("Done!") })
        )

        return inflater.inflate(R.layout.fragment_recipe_list, container, false)
    }

    override fun onStop() {
        super.onStop()
        mDisposable.dispose()
    }
}
